core/test/concatenate_test.cpp

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+#include "test_macros.h"
+#include <gtest/gtest.h>
+
+#include "scipp/core/dataset.h"
+
+using namespace scipp;
+using namespace scipp::core;
+
+TEST(ConcatenateTest, simple_1d) {
+  Dataset a;
+  a.setCoord(Dim::X, makeVariable<int>({Dim::X, 3}, {1, 2, 3}));
+  a.setData("data_1", makeVariable<int>({Dim::X, 3}, {11, 12, 13}));
+
+  Dataset b;
+  b.setCoord(Dim::X, makeVariable<int>({Dim::X, 3}, {4, 5, 6}));
+  b.setData("data_1", makeVariable<int>({Dim::X, 3}, {14, 15, 16}));
+
+  const auto d = concatenate(a, b, Dim::X);
+
+  EXPECT_EQ(makeVariable<int>({Dim::X, 6}, {1, 2, 3, 4, 5, 6}),
+            d.coords()[Dim::X]);
+  EXPECT_EQ(makeVariable<int>({Dim::X, 6}, {11, 12, 13, 14, 15, 16}),
+            d["data_1"].data());
```

Comment:
Generally we are using `ASSERT(actual, expected)` throughout scipp (not the other way round), could you adopt that convention?